### PR TITLE
non_indexed_labels stage

### PR DESCRIPTION
**Pull request by vlad-diachenko**
#### PR Description

Added non_indexed_labels stage that attaches the label<>values from extracted data to log line entry.

#### Which issue(s) this PR fixes

it repeats the implementation added to promtail in https://github.com/grafana/loki/pull/9986

#### Notes to the Reviewer
Documentation will be added in a separate PR

#### PR Checklist

- [x] CHANGELOG updated
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Main (unreleased)
   - `discovery.eureka` discovers targets from a Eureka Service Registry. (@spartan0x117)
   - `discovery.openstack` - service discovery for OpenStack. (@marctc)
   - `loki.write` now exposes basic WAL support. (@thepalbi)
+  - `loki.process` - added a new stage `non_indexed_labels` to attach non-indexed labels from extracted data to log line entry. (@vlad-diachenko)
```
Comment (rfratto) [inline CHANGELOG.md:38]: This wouldn't be listed as a new component, but rather an enhancement to an existing component.

File: go.mod [hunk 2]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/Shopify/sarama v1.38.1
 	github.com/alecthomas/kingpin/v2 v2.3.2
 	github.com/alecthomas/units v0.0.0-20211218093645-b94a6e3cc137
-	github.com/aws/aws-sdk-go v1.44.304
+	github.com/aws/aws-sdk-go v1.44.311
```
Comment (rfratto) [inline go.mod:16]: Do we need to be updating any of our dependencies in this PR? I'm particularly worried about the blast radius of modifying pkg/agentproto.

File: CHANGELOG.md (hunk 1 above)
Comment (rfratto) [inline CHANGELOG.md:38]: Can you make sure to update the reference documentation page for loki.process with the new stage?

File: component/loki/process/stages/nonindexedlabels.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func (s *nonIndexedLabelsStage) Name() string {
+	return StageTypeNonIndexedLabels
+}
+
+func (s *nonIndexedLabelsStage) Run(in chan Entry) chan Entry {
+	return RunWith(in, func(e Entry) Entry {
+		processLabelsConfigs(s.logger, e.Extracted, s.cfgs, func(labelName model.LabelName, labelValue model.LabelValue) {
+			e.NonIndexedLabels = append(e.NonIndexedLabels, logproto.LabelAdapter{Name: string(labelName), Value: string(labelValue)})
```
[the remark this answers is not in the document]
Comment (vlad-diachenko) [reply component/loki/process/stages/nonindexedlabels.go:33]: it is a new feature added to Loki. once it's fully released, the users will automatically get this non-indexed labels extracted even without running any parsers like `| json` or `| logfmt`.... 
here is the docs we added in loki https://github.com/grafana/loki/pull/10073/files

File: go.mod (hunk 2 above)
Comment (vlad-diachenko) [reply go.mod:16]: I had to update the version of these 3 dependencies:
1. github.com/grafana/loki
2. github.com/grafana/loki/pkg/push - to have the new request proto model
3. github.com/grafana/windows_exporter - the previous version did not exist in the repo, so, I failed to run `go get ....` or `go mod tidy` . 

and these updates bring some other dependencies updates.

**Review by clayton-cornell — COMMENTED** [review]
Docs look OK as-is

File: CHANGELOG.md (hunk 1 above)
Comment (vlad-diachenko) [reply CHANGELOG.md]: done

Comment (vlad-diachenko) [reply CHANGELOG.md]: updated the docs

File: docs/sources/flow/reference/components/loki.process.md [hunk 4]
```diff
@@ -330,6 +330,30 @@ stage.labels {
...
+The `stage.non_indexed_labels` inner block configures a labels processing stage that can read
+data from the extracted values map and add them to log entries as non-indexed labels.
+
+The following arguments are supported:
+
+| Name     | Type          | Description                                                                 | Default | Required |
+| -------- | ------------- |-----------------------------------------------------------------------------| ------- | -------- |
+| `values` | `map(string)` | Specifies the list of labels to add from extracted values map to log entry. | `{}`    | no       |
+
+In a labels stage, the map's keys define the label to set and the values are
```
Comment (thampiotr) [inline docs/sources/flow/reference/components/loki.process.md:344]:
```suggestion
In a non_indexed_labels stage, the map's keys define the label to set and the values are
```

File: docs/sources/flow/reference/components/loki.process.md [hunk 5]
```diff
@@ -330,6 +330,30 @@ stage.labels {
...
+how to look them up. If the value is empty, it is inferred to be the same as
+the key.
+
+```river
+stage.labels {
```
Comment (thampiotr) [inline docs/sources/flow/reference/components/loki.process.md:349]:
```suggestion
stage.non_indexed_labels {
```

File: docs/sources/flow/reference/components/loki.process.md [hunk 6]
```diff
@@ -330,6 +330,30 @@ stage.labels {
 }
 ```
 
+### stage.non_indexed_labels block
+
+The `stage.non_indexed_labels` inner block configures a labels processing stage that can read
+data from the extracted values map and add them to log entries as non-indexed labels.
+
```
Comment (thampiotr) [inline docs/sources/flow/reference/components/loki.process.md:337]: Could you explain here what the non-indexed labels and refer to Loki's docs for further reading?

Comment (vlad-diachenko) [reply docs/sources/flow/reference/components/loki.process.md:337]: Yes, but Loki docs are not published yet. once it's published I will add the link to the page

**Review by thampiotr — APPROVED** [review]
LGTM % the docs explaining more what the non_indexed_labels are and linking to Loki docs once merged :)  

Thanks for adding the extra auth tests!

**Review by clayton-cornell — APPROVED** [review]

File: docs/sources/flow/reference/components/loki.process.md (hunk 5 above)
Comment (thampiotr) [reply docs/sources/flow/reference/components/loki.process.md]: I think this still needs to be done?

File: docs/sources/flow/reference/components/loki.process.md (hunk 4 above)
Comment (thampiotr) [reply docs/sources/flow/reference/components/loki.process.md]: This is not a `labels` stage we're talking about here, but `non_indexed_labels` stage, right?

**Review by thampiotr — COMMENTED** [review]
~I've noticed that a couple of comments may still need addressing?~
Sorry, had a wrong URL pinning to old version of this PR. Ignore this.

**Pull request merged**